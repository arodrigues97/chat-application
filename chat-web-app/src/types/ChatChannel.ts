export type ChatChannel = {
  id: string
  name: string
}
