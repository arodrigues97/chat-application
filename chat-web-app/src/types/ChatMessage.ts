import { User } from "./User"

export type ChatMessage = {
  id: string
  message: string
  sender: User
  timeStamp: Date
}
