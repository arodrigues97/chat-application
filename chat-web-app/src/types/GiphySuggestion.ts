export type GiphySuggestion = {
  id: string
  embedUrl: string
  title: string
}
