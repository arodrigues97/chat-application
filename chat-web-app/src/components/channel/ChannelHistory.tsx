import { Header, Icon, Loader, Segment } from "semantic-ui-react"
import { ChatChannel } from "../../types/ChatChannel"
import ChatBubble from "../chat/ChatBubble"

export type ChannelHistoryProps = {
  channel: ChatChannel
}

const ChannelHistory = (props: ChannelHistoryProps) => {
  const { channel } = props
  const { messages } = channel
  return (
    <div>
      {messages && messages.length === 0 && (
        <Segment placeholder>
          <Header icon>
            <Icon name="exclamation" />
            No messages
          </Header>
        </Segment>
      )}
      <ul className="channelMessageList">
        {messages &&
          messages.length > 0 &&
          messages.map((message) => (
            <li key={message.id}>
              <ChatBubble message={message} />
            </li>
          ))}
      </ul>
    </div>
  )
}

export default ChannelHistory
