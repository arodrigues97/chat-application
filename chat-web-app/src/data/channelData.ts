import { ChatChannel } from "../types/ChatChannel"

//Mock data while preparing the front-end
export const channelData: ChatChannel[] = [
  {
    id: "32323",
    name: "Homestars",
  },
  {
    id: "34343",
    name: "Adam Rodrigues",
  },
]
