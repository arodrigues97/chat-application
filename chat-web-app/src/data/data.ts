import { ChatChannel } from "../types/ChatChannel"
import { User } from "../types/User"

/**
 * Mock information if opting not use the API
 */

export const userData: User[] = [
  {
    id: 1,
    name: "Adam Rodrigues",
  },
  {
    id: 2,
    name: "Ankit Kanojia",
  },
]

export const mockUser = userData[0]

export const channelData: ChatChannel[] = [
  {
    id: 1,
    name: "Homestars",
    users: [userData[0], userData[1]],
    messages: [
      {
        id: 1,
        channelId: 1,
        user: userData[0],
        timeStamp: "2021-07-22 11:20 am",
        message: "Hello, everyone. It's me Adam Rodrigues.",
      },
      {
        id: 2,
        channelId: 1,
        user: userData[0],
        timeStamp: "2021-07-22 11:21 am",
        message: "Hello, everyone. It's me Adam Rodrigues.",
      },
    ],
  },
]
