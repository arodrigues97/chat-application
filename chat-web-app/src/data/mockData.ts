import { ChatChannel } from "../types/ChatChannel"
import { User } from "../types/User"

export const userData: User[] = [
  {
    id: 1,
    name: "Adam Rodrigues",
  },
  {
    id: 2,
    name: "Ankit Kanojia",
  },
]

export const mockUser = userData[0]

export const channelData: ChatChannel[] = [
  {
    id: 1,
    name: "Homestars",
    users: [userData[0], userData[1]],
    messages: [
      {
        id: 1,
        channelId: 1,
        user: userData[0],
        timeStamp: "2021-07-22 11:20 am",
        message: "Hello, everyone. It's me Adam Rodrigues.",
      },
      {
        id: 2,
        channelId: 1,
        user: userData[0],
        timeStamp: "2021-07-22 11:21 am",
        message: "Hello, everyone. It's me Adam Rodrigues.",
      },
    ],
  },
  {
    id: 2,
    name: "Adam's Channel",
    users: [userData[0], userData[1]],
    messages: [
      {
        id: 3,
        channelId: 2,
        user: userData[0],
        timeStamp: "2021-07-22 11:20 am",
        message: "Hello",
      },
      {
        id: 4,
        channelId: 2,
        user: userData[0],
        timeStamp: "2021-07-22 11:21 am",
        message: "Test.",
      },
    ],
  },
]
