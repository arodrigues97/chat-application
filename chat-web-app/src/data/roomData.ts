import { ChatRoom } from "../types/ChatRoom"

//Mock data while preparing the front-end
export const rooms: ChatRoom[] = [
  {
    id: "32323",
    name: "Homestars Room",
  },
  {
    id: "34343",
    name: "Adam Rodrigue's Room",
  },
]
