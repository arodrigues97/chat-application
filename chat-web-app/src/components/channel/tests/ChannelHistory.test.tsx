import { render, screen } from "@testing-library/react"
import { ChangeEvent } from "react"
import { channelData, mockUser } from "../../../data/mockData"
import { ChatMessage } from "../../../types/ChatMessage"
import ChannelHistory from "../ChannelHistory"
import { mount } from "enzyme"

describe("Channel History Tests", () => {
  let activeChannel = channelData[0]
  let wrapper: any
  beforeEach(() => {
    wrapper = render(
      <ChannelHistory
        channel={activeChannel}
        user={mockUser}
        editMessage={""}
        handleEditMessageChange={function (
          event: ChangeEvent<HTMLInputElement>
        ): void {
          throw new Error("Function not implemented.")
        }}
        handleEditMessageSave={function (message: ChatMessage): void {
          throw new Error("Function not implemented.")
        }}
        handleToggleEditMessage={function (message: ChatMessage): void {
          throw new Error("Function not implemented.")
        }}
        handleDeleteMessage={function (message: ChatMessage): void {
          throw new Error("Function not implemented.")
        }}
      />
    )
  })

  it("should render", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("renders messages", () => {
    const { getAllByText } = screen
    activeChannel.messages.forEach((message) => {
      expect(getAllByText(message.message)).toBeDefined()
    })
  })

  it("no messages text", () => {
    render(
      <ChannelHistory
        channel={{
          ...activeChannel,
          messages: [],
        }}
        user={mockUser}
        editMessage={""}
        handleEditMessageChange={function (
          event: ChangeEvent<HTMLInputElement>
        ): void {
          throw new Error("Function not implemented.")
        }}
        handleEditMessageSave={function (message: ChatMessage): void {
          throw new Error("Function not implemented.")
        }}
        handleToggleEditMessage={function (message: ChatMessage): void {
          throw new Error("Function not implemented.")
        }}
        handleDeleteMessage={function (message: ChatMessage): void {
          throw new Error("Function not implemented.")
        }}
      />
    )
    const { getByText } = screen
    expect(getByText("No messages"))
  })
})
