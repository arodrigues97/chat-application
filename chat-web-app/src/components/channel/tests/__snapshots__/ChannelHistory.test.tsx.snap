// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel History Tests should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="chatHistory"
      >
        <ul
          class="channelMessageList"
        >
          <li>
            <div
              class="ui message"
            >
              <i
                aria-hidden="true"
                class="close icon"
              />
              <div
                class="header"
              >
                Adam Rodrigues
              </div>
              <div
                class="content chatBubble"
              >
                <div
                  class="message"
                >
                  Hello, everyone. It's me Adam Rodrigues.
                </div>
                <div
                  class="timestamp"
                >
                  2021-07-22 11:20 am
                </div>
              </div>
              <i
                aria-hidden="true"
                class="edit link icon"
                id="edit-icon"
              />
            </div>
          </li>
          <li>
            <div
              class="ui message"
            >
              <i
                aria-hidden="true"
                class="close icon"
              />
              <div
                class="header"
              >
                Adam Rodrigues
              </div>
              <div
                class="content chatBubble"
              >
                <div
                  class="message"
                >
                  Hello, everyone. It's me Adam Rodrigues.
                </div>
                <div
                  class="timestamp"
                >
                  2021-07-22 11:21 am
                </div>
              </div>
              <i
                aria-hidden="true"
                class="edit link icon"
                id="edit-icon"
              />
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chatHistory"
    >
      <ul
        class="channelMessageList"
      >
        <li>
          <div
            class="ui message"
          >
            <i
              aria-hidden="true"
              class="close icon"
            />
            <div
              class="header"
            >
              Adam Rodrigues
            </div>
            <div
              class="content chatBubble"
            >
              <div
                class="message"
              >
                Hello, everyone. It's me Adam Rodrigues.
              </div>
              <div
                class="timestamp"
              >
                2021-07-22 11:20 am
              </div>
            </div>
            <i
              aria-hidden="true"
              class="edit link icon"
              id="edit-icon"
            />
          </div>
        </li>
        <li>
          <div
            class="ui message"
          >
            <i
              aria-hidden="true"
              class="close icon"
            />
            <div
              class="header"
            >
              Adam Rodrigues
            </div>
            <div
              class="content chatBubble"
            >
              <div
                class="message"
              >
                Hello, everyone. It's me Adam Rodrigues.
              </div>
              <div
                class="timestamp"
              >
                2021-07-22 11:21 am
              </div>
            </div>
            <i
              aria-hidden="true"
              class="edit link icon"
              id="edit-icon"
            />
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
